pkg/destroy/aws/aws.go

```diff
@@ -77,1510 +63,1205 @@ func (o *ClusterUninstaller) validate() error {
+	response, err := client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: []*string{aws.String(id)},
 	})
 	if err != nil {
 		return err
 	}
 
-	for _, policy := range results.PolicyNames {
-		_, err := iamClient.DeleteRolePolicy(&iam.DeleteRolePolicyInput{
-			RoleName:   role,
-			PolicyName: policy,
-		})
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// deleteRolesFromInstanceProfile will attempt to delete any roles associated with a given instance profile
-func deleteRolesFromInstanceProfile(ip *iam.InstanceProfile, iamClient *iam.IAM, logger logrus.FieldLogger) error {
-	for _, role := range ip.Roles {
-		logger.Debugf("deleting role %v from instance profile %v", *role.RoleName, *ip.InstanceProfileName)
-
-		// empty the role
-		logger.Debugf("deleting policies from role: %v", *role.RoleName)
-		err := deletePoliciesFromRole(role.RoleName, iamClient)
-		if err != nil {
-			logger.Debugf("error deleting policies from role: %v", err)
-			return err
-		}
-
-		logger.Infof("Deleted all policies from role: %v", *role.RoleName)
-
-		// detach role from instance profile
-		_, err = iamClient.RemoveRoleFromInstanceProfile(&iam.RemoveRoleFromInstanceProfileInput{
-			InstanceProfileName: ip.InstanceProfileName,
-			RoleName:            role.RoleName,
-		})
-		if err != nil {
-			logger.Debugf("error removing role from instance profile: %v", err)
-			return err
+	for _, group := range response.SecurityGroups {
+		if len(group.IpPermissions) > 0 {
+			_, err := client.RevokeSecurityGroupIngress(&ec2.RevokeSecurityGroupIngressInput{
+				GroupId:       group.GroupId,
+				IpPermissions: group.IpPermissions,
+			})
+			if err != nil {
+				return errors.Wrap(err, "revoking ingress permissions")
+			}
+			logger.Debug("Revoked ingress permissions")
 		}
 
-		logger.Infof("Removed role %v from instance profile %v", *role.RoleName, *ip.InstanceProfileName)
-
-		// now delete the role
-		// need to loop because this is the only time we'll have the name of the role
-		// now that it has been detached from the instance profile
-		for {
-			_, err = iamClient.DeleteRole(&iam.DeleteRoleInput{
-				RoleName: role.RoleName,
+		if len(group.IpPermissionsEgress) > 0 {
+			_, err := client.RevokeSecurityGroupEgress(&ec2.RevokeSecurityGroupEgressInput{
+				GroupId:       group.GroupId,
+				IpPermissions: group.IpPermissionsEgress,
 			})
 			if err != nil {
-				logger.Debugf("error deleting role %v from instance profile %v: %v", *role.RoleName, ip.InstanceProfileName, err)
-			} else {
-				logger.WithField("name", *role.RoleName).Info("Deleted role")
-				break
+				return errors.Wrap(err, "revoking egress permissions")
 			}
+			logger.Debug("Revoked egress permissions")
+		}
+	}
 
-			time.Sleep(time.Second * secondsToSleep)
+	_, err = client.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
+		GroupId: aws.String(id),
+	})
+	if err != nil {
+		if err.(awserr.Error).Code() == "InvalidGroup.NotFound" {
```

Comment:
Is there no constant for `NotFound` ?

Reply:
> Is there no constant for `NotFound` ?

Not that I could find.  And Terraform is also using these strings:

```console
$ git grep InvalidGroup.NotFound
pkg/destroy/aws/aws.go:         if err.(awserr.Error).Code() == "InvalidGroup.NotFound" {
pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_autoscaling_group.go:                          case "InvalidGroup.NotFound":
pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_autoscaling_group.go:          if ok && autoscalingerr.Code() == "InvalidGroup.NotFound" {
pkg/terraform/exec/plugins/vendor/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_elb.go:                                ec2err.Code() == "InvalidGroup.NotFound" {
...
```

Reply:
:+1: